pycontrails/core/met.py

```diff
@@ -97,17 +97,20 @@ def _validate_dim_contains_coords(self) -> None:
         ValueError
             If data does not contain all four coordinates (longitude, latitude, level, time).
         """
-        for dim in self.dim_order:
-            if dim not in self.data.dims:
-                if dim == "level":
-                    msg = (
-                        f"Meteorology data must contain dimension '{dim}'. "
-                        "For single level data, set 'level' coordinate to constant -1 "
-                        "using `ds = ds.expand_dims({'level': [-1]})`"
-                    )
-                else:
-                    msg = f"Meteorology data must contain dimension '{dim}'."
-                raise ValueError(msg)
+        missing = set(self.dim_order).difference(self.data.dims)
+        if not missing:
+            return
+
+        dim = sorted(missing)[0]
+        if dim == "level":
+            msg = (
+                f"Meteorology data must contain dimension '{dim}'. "
+                "For single level data, set 'level' coordinate to constant -1 "
+                "using `ds = ds.expand_dims({'level': [-1]})`"
+            )
+        else:
+            msg = f"Meteorology data must contain dimension '{dim}'."
```

Comment:
```suggestion
        dim = sorted(missing)
        dim_str = ", ".join(f"'{dim}'")
        msg = f"Meteorology data must contain dimension(s) {dim_str}."
        if "level" in dim:
           msg += (
               "For single level data, set 'level' coordinate to constant -1 "
                "using `ds = ds.expand_dims({'level': [-1]})`"
           )
```
(Feel free to take or leave this--might be nice for users to get told about all missing dimensions at once.)

Reply:
Thank you! This looks great.